torch/jit/__init__.py

```diff
@@ -1091,6 +1091,8 @@ def __setattr__(self, attr, value):
                 super(ScriptModule, self).__setattr__(attr, _ConstModuleList(value))
             elif isinstance(value, Sequential):
                 super(ScriptModule, self).__setattr__(attr, _ConstSequential(value))
+            elif isinstance(value, Parameter):
```

Comment:
If you move the `# Copy constants` to after the parameters get copied from the original to the weak module you should be able to get rid of this case (since they wouldn't be in `self._constants_set` yet)